quality_control_mrp/models/mrp_production.py

```diff
@@ -24,13 +24,15 @@ def _count_inspections(self):
     @api.model
     def action_produce(self, production_id, production_qty, production_mode,
                        wiz=False):
+        production = self.browse(production_id)
+        done_moves = production.move_created_ids2.filtered(
+            lambda r: r.state == 'done')
         res = super(MrpProduction, self).action_produce(
             production_id, production_qty, production_mode, wiz=wiz)
         if production_mode == 'consume_produce':
             inspection_model = self.env['qc.inspection']
-            production = self.browse(production_id)
             for move in production.move_created_ids2.filtered(
-                    lambda r: r.state == 'done'):
+                    lambda r: r.state == 'done' and r not in done_moves):
```

Comment:
Better to use difference operator: `for move in production.move_created_ids2.filtered(lambda r: r.state == 'done') - done_moves:`